### ISSUE TITLE
Provide a Community Development Server for Wechaty Contributors

### ISSUE THREAD

**Issue by huan** [wechaty/PMC#13]
In order to help our contributors to develop and test our Wechaty Open Source Software with convenience, especially for our Hostie Service with GRPC, Multi-language Wechaty, and Plugins, we decide to provide a public Ubuntu server to serving our Wechaty Contributors.

### The Plan

For now, our plan is:

1. The server will be located in the South East of Asia, which has a little network latency but lives out of GFW.
1. The server is running Ubuntu 18.04 LTS Linux
1. The server will have 4/core CPU, 16GB ram memory with 1TB SSD at `/home` for sharing.
1. The username for all Wechaty Contributors will be the same as the GitHub username
1. All contributors need to use their own SSH key to log in to the server, and the SSH key must be registered on GitHub, which can be read by the GitHub API: [List public keys for a user](https://developer.github.com/v3/users/keys/). (e.g. SSH Public Keys for me is <https://api.github.com/users/huan/keys>)

### Cloud Server ($)

The Wechaty Contributor Server is currently hosted on [Microsoft Azure Cloud](https://azure.com) at [Asia Southeast Region](https://azure.microsoft.com/en-us/global-infrastructure/geographies/)

| Resource | Size | Cost ($ per month) |
| ---: | :--- | :--- |
| [Managed Disk](https://azure.microsoft.com/en-us/pricing/details/managed-disks/) | 1TiB (Premium SSD) | $135.17 |
| [Virtual Machine](https://docs.microsoft.com/en-us/azure/virtual-machines/sizes) | 16 GiB Ram, 4 vCPU (Standard D4s v3) | $182.50 |
| Total | : | $317.67 |

> Azure Cloud sponsored by [Microsoft MVP](https://mvp.microsoft.com/) Sponsorship.

### Code of Conduct

The Code of Conduct for using our Wechaty Contributor Server are:

1. MUST not break the law
1. MUST not make other contributors un-happy
1. MUST not run any task unrelated to Wechaty Open Source Software Project. (Includes but not limited to VPN, Mining, Attacking)
1. MUST not upload sensitive data. The server is public for all contributors, so there's only public data that can be put on the server.

Anyone, no matter who breaks the Code of Conduct will be removed from the server.

### The Service Level Agreements (SLA)

We do not have any SLA for our Wechaty Contributors Server.

1. The server can be rebooted or reinstalled at any time
1. The data may be lost at any time
1. The service may be stopped permanently at any time.

So please use it at your own risk.

### As-Is - No Warranty

- Server Address: `dev.chatie.io`
- Login Protocol: SSH
- User Name: _The Same As Your GitHub UserName_
- Password: _Not Supported. Use Your SSH Key Registered with the GitHub_

I hope this server will helpful for our community, please feel free to let me know if you have any suggestion/questions by commenting on this issue.

Cheers!

Huan

**Comment by huan** [comment]
### Setup Script for Azure Ubuntu 18.04 LTS

```sh
sudo apt update
sudo apt install python3.8 python3.8-dev
sudo apt install python3-pip python-pip
sudo pip3 install --upgrade netifaces
sudo update-alternatives --config python3

sudo apt remove python3-apt
sudo apt install python3-apt
```

#### Node.js

```sh
curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -
sudo apt-get install -y nodejs
```

#### Swap File

```sh
sudo dd if=/dev/zero of=/swapfile bs=1024 count=4096000
sudo chmod 0600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile

sudo cat <<_EOF_ > /etc/rc.local
#!/usr/bin/env bash
swapon /swapfile
_EOF_
sudo chmod +x /etc/rc.local
```

**Comment by lucifer1004** [comment]
### Apply for docker access

Usage: gateway server

**Comment by huan** [comment]
@lucifer1004 Approved. You have been added to the `docker` user group. Please relogin and confirm your group by run `id`

**Comment by fangmuke** [comment]
申请Docker Run权限👀

**Comment by huan** [comment]
@fangmuke Approved. You have been added to the docker user group. Please relogin and confirm your group by run `id`

**Comment by ly1ex** [comment]
Apply for docker access
Usage: padlocal server

**Comment by huan** [comment]
@lyleshaw Approved. You need to log out and log in again to update your group id, please let me know it works or not.

BTW: I noticed that you have a great project at https://github.com/lyleshaw/WeChaty-Torch , it would be great if you can write a blog post to share it to the community.

**Comment by ly1ex** [comment]
> @lyleshaw Approved. You need to log out and log in again to update your group id, please let me know it works or not.
> 
> BTW: I noticed that you have a great project at https://github.com/lyleshaw/WeChaty-Torch , it would be great if you can write a blog post to share it to the community.

Thank you very much, now docker can be used.
And I have posted a blog about [https://github.com/lyleshaw/WeChaty-Torch](https://github.com/lyleshaw/WeChaty-Torch) before, you can read it [here](https://wechaty.js.org/2020/10/25/use-wechaty-to-predict-image/).I will improve this project and rewrite a blog to introduce the details when I have time~

**Comment by osindex** [comment]
Apply for docker access
Usage: puppet server test

**Comment by huan** [comment]
@osindex Applied.

**Comment by AntonioSun** [comment]
Apply for docker access
Usage: padlocal server & puppet-discord

**Comment by huan** [comment]
@AntonioSun It seems that you are not a member of the Wechaty organization yet.

Please join the [Wechaty Contributor Program](https://wechaty.js.org/docs/contributor-program/) if you have not yet.

**Comment by suntong** [comment]
Apply for docker access
Usage: padlocal server & puppet-discord

**Comment by huan** [comment]
@suntong Of course. Have added you to the `docker` group, please re-login and then you will have the docker access.

**Comment by suntong** [comment]
TY sir.

**Comment by lhr0909** [comment]
@huan I would like to get docker access for setting up a Rasa NLU endpoint for you 😄

**Comment by huan** [comment]
@lhr0909 Of course, have added you to the `docker` group.

```sh
lhr0909@wechaty-contributors:~$ id
uid=1009(lhr0909) gid=1009(lhr0909) groups=1009(lhr0909),116(docker),1001(wechaty)
```

Thank you very much for helping the community for setting up the Rasa NLU endpoint!

**Comment by huan** [comment]
# 📢 Announcement: Sunsetting the Wechaty Community Development Server

Dear Wechaty community,

Since 2020, our **Community Development Server** has been sponsored and maintained as a free shared resource for active contributors (see the original PMC proposal for details). It has helped many of us prototype bots, test new ideas, and collaborate more easily.

However, after careful consideration of **costs, maintenance effort, and security responsibilities**, the PMC and VHIT community have decided to **retire the Community Development Server**.

---

## 🕒 Timeline

* **Now → December 31, 2025**
  The Community Development Server will continue running during this period. You may still access your environment, but **you should treat it as read‑only / migration phase** and start preparing your own infrastructure.

* **After December 31, 2025**
  We will begin the shutdown process.

* **Starting from January 1, 2026**
  The **Wechaty Community Development Server will be permanently closed**. All services running on it will be stopped, and the server will no longer be available.

---

## 📦 What You Need to Do

If you currently have anything on the Community Development Server, please **migrate as soon as possible**:

* ✅ **Back up your data** (code, logs, configuration files, databases, etc.)
* ✅ **Migrate any running services** (bots, webhooks, dashboards, demos, etc.) to your own:

  * Cloud VM (e.g. GCP, Azure, AWS, etc.)
  * Self‑hosted server
  * Other hosting providers (Render, Railway, Fly.io, etc.)
* ✅ **Update your webhooks, tokens, environment variables, and DNS** to point to your new infrastructure

After the shutdown date, **you will not be able to access the server anymore**.

---

## ⚠️ Important Disclaimer

Starting from **January 1, 2026**:

* The **Wechaty PMC, Wechaty community, and related maintainers will not be responsible** for:

  * Any **data loss** on the Community Development Server
  * Any **service downtime** or impact on individual or business users caused by the shutdown
* The Community Development Server is provided **“as is”** during the remaining time, **without any uptime guarantees**.

By continuing to use the server during the transition period, you acknowledge and accept the above terms.

---

## 💚 Thank You

We want to sincerely **thank all contributors** who have used, improved, and supported the Community Development Server over the years. It has been an important part of our community’s early growth.

As we move forward, we encourage everyone to:

* Keep building amazing things with **Wechaty** 💬🤖
* Share your setups, best practices, and deployment guides with the community
* Continue collaborating via GitHub, the Wechaty docs, and our community channels

If you have any questions about this announcement or need help migrating your workloads, please **open an issue in the `wechaty/PMC` repo** or ask in our community channels.

On behalf of the VHIT community and the Wechaty PMC,

**Wechaty Project Management Committee (PMC)**
